daemon/containerd/image_delete.go

```diff
@@ -127,7 +141,16 @@ func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, options
 
 		using := func(c *container.Container) bool {
 			if c.ImageID == imgID {
-				return true
+				if len(options.Platforms) > 0 {
+					for _, p := range options.Platforms {
+						pm := platforms.OnlyStrict(p)
+						if pm.Match(c.ImagePlatform) {
+							return true
+						}
+					}
```

Comment:
Perhaps this path could use a comment to outline we're intentionally _not_ returning `false` after this, something like;

```go
// No match for the image reference, but continue to check if used as mounted image
```

(better suggestions welcome!)


(because I was wondering if we forgot a `return false` here; the comment may help future visitors 😄

Reply:
☝️ really wondering if this function needs splitting up, because there's a _ton_ of conditions (not for this PR!)

Reply:
Yes actually I started by refactoring it to make the flow more linear, but turned out that I broke some edge cases in the process and didn't want to spend more time on that 🙈